File: paleo_workbench/ui/pages/message_preview_widget.py
from __future__ import annotations

from PySide6.QtCore import Qt
from PySide6.QtWidgets import QLabel


class MessagePreviewWidget(QLabel):
    def __init__(self, parent=None):
        super().__init__(parent)
        self.setAlignment(Qt.AlignmentFlag.AlignCenter)
        self.setWordWrap(True)

    def set_message(self, text: str) -> None:
        self.setText(text)

File: libs/ui_pages_preview/src/qt/message_preview_widget.cpp
#include <pwb/ui_pages_preview/qt/message_preview_widget.hpp>

namespace pwb::ui_pages_preview {

MessagePreviewWidget::MessagePreviewWidget(QWidget* parent)
    : QLabel(parent) {
    setAlignment(Qt::AlignCenter);
    setWordWrap(true);
}

}  // namespace pwb::ui_pages_preview
